src/caffe/solver.cpp

```diff
@@ -469,6 +495,32 @@ void SGDSolver<Dtype>::ApplyUpdate() {
 }
 
 template <typename Dtype>
+void SGDSolver<Dtype>::Normalize(int param_id) {
+  if (this->param_.iter_size() == 1) { return; }
+  // Scale gradient to counterbalance accumulation.
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
+  const Dtype accum_normalization = Dtype(1.) / this->param_.iter_size();
```

Comment:
It is done this way due to the separation of `Net` and `Solver` but it is correct. `Net` normalizes by the (computation) batch size but only `Solver` knows about `iter_size` so it does the portion of the normalization needed to handle accumulation.